TFNetworkRecLayer.py

```diff
@@ -4134,12 +4135,16 @@ def get_rec_initial_extra_outputs(cls, batch_dim, rec_layer, num_heads, total_ke
     data = get_concat_sources_data_template(sources)
     data = data.copy_as_batch_major()
     if data.time_dim_axis is None:
+      kv_dim = total_key_dim + n_out
       # Assume inside RecLayer.
       # Before, we used a tf.TensorArray.
       # However, that has higher memory consumptions than just using a tensor and concatenating to it.
-      total_value_dim = n_out
       # (batch,heads,time,kv-dim//heads)
-      kv_left = tf.zeros((batch_dim, num_heads, 0, (total_key_dim + total_value_dim) // num_heads))
+      kv_left = RnnCellLayer.get_rec_initial_state_inner(rec_layer=rec_layer, state_key="kv_left",
+                                                         name=kwargs['name'],
+                                                         initial_state=kwargs['initial_state'],
+                                                         initial_shape=(batch_dim, num_heads, 0, kv_dim // num_heads),
```

Comment:
`initial_state` is an optional argument, or not? So it would be `kwargs.get('initial_state', None)`.
But better, why do you access `kwargs` at all here? Just make it an explicit arg of the method `get_rec_initial_extra_outputs`, with the same default as in `__init__`.
The same also for `name`.

Reply:
Changed accordingly. Thanks!